src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -93,6 +97,42 @@ def _dist_count_aggregation_on_tx_status_factory(
     )
 
 
+def _aggregation_on_tx_satisfaction_func_factory(aggregate):
```

Comment:
We probably can have a single factory handling what this and the factory above do. I will open a PR in the future to merge these two.